File: devo-front/src/app/services/models/purchaseRequest.ts
export enum purchaseStatus {
  New = 'New',
  Resolved = 'Resolved',
  InProgress = 'InProgress',
  Completed = 'Completed'
}

export interface purchaseRequest {
     purchaseid?: number;
     purchasedate?: string;
    description?: string;
  purchaseStatus?: purchaseStatus;
}

File: gcp/chatintell/src/main/java/com/example/chatintell/entity/PurchaseRequest.java
package com.example.chatintell.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Table(name = "purchaserequest")
public class PurchaseRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer purchaseid;
    private LocalDate purchasedate;
    private String createdBy;
    @Column(columnDefinition = "TEXT")
    private String description;
    @Enumerated(EnumType.STRING)
    private PurchaseStatus purchaseStatus;
    @ManyToOne
    private PaymentOrder paymentorder;
    @ManyToOne
    private Ticket ticket;





}
